numpy/random/src/philox/philox.h

```diff
@@ -32,6 +32,17 @@ static NPY_INLINE uint64_t mulhilo64(uint64_t a, uint64_t b, uint64_t *hip) {
   *hip = product >> 64;
   return (uint64_t)product;
 }
+#elif defined(_M_X64) || defined(_M_ARM64)
+#include <intrin.h>
+#pragma intrinsic(__umulh)
+static NPY_INLINE uint64_t mulhilo64(uint64_t a, uint64_t b, uint64_t *hip) {
+  uint64_t p0;
+  uint64_t p1;
+  p0 = a * b;
+  p1 = __umulh(a, b);
+  *hip = p1;
+  return p0;
```

Comment:
nit:
```suggestion
  uint64_t p0 = a * b;
  uint64_t p1 = __umulh(a, b);
  *hip = p1;
  return p0;
```
or if you don't think it harms clarity
```suggestion
  *hip = __umulh(a, b);
  return a * b;
```

Reply:
I would like to keep the p0, p1 namings, as the compiler will make the correct optimizations and the same jargon is used in the documentation I found available.